From 9f3c31cd8d68b5272f803063247362a644720227 Mon Sep 17 00:00:00 2001
From: Kairui Song <kasong@redhat.com>
Date: Mon, 20 Aug 2018 18:43:19 +0800
Subject: [PATCH] 99base: enable initqueue if extra devices are added

When extra devices are added, initqueue should be enabled to make sure
those devices are present, so following services and routines could
use those devices.

See PR #442 for more detail.
---
 modules.d/99base/module-setup.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules.d/99base/module-setup.sh b/modules.d/99base/module-setup.sh
index a1569b12..731c6dcc 100755
--- a/modules.d/99base/module-setup.sh
+++ b/modules.d/99base/module-setup.sh
@@ -94,6 +94,9 @@ install() {
 
     ## save host_devs which we need bring up
     if [[ $hostonly_cmdline == "yes" ]]; then
+        if [[ -n $add_device ]]; then
+            dracut_need_initqueue
+        fi
         if [[ -f "$initdir/lib/dracut/need-initqueue" ]] || ! dracut_module_included "systemd"; then
             (
                 if dracut_module_included "systemd"; then

